Packet++/header/StpLayer.h

```diff
@@ -0,0 +1,834 @@
+		 * @return uint8_t Flags of the configuration message
+		 */
+		inline uint8_t getFlag() const { return getStpConfHeader()->flag; }
+
+		/**
+		 * Returns the root bridge identifier
+		 * @return uint64_t root bridge identifier
+		 */
+		inline uint64_t getRootId() const { return be64toh(getStpConfHeader()->rootId); }
+
+		/**
+		 * Returns the priority of root bridge
+		 * @return uint16_t Priority of root bridge
+		 */
+		inline uint16_t getRootPriority() const { return be16toh(getStpConfHeader()->rootId) & 0xf000; }
+
+		/**
+		 * Returns the system identifier extension of root bridge
+		 * @return uint16_t System extension of root bridge
+		 */
+		inline uint16_t getRootSystemIDExtension() const { return be16toh(getStpConfHeader()->rootId) & 0x0fff; }
+
+		/**
+		 * Returns the system identifier of root bridge
+		 * @return pcpp::MacAddress System identifier of root bridge
+		 */
+		inline pcpp::MacAddress getRootSystemID() const { return IDtoMacAddress(getRootId()); }
+
+		/**
+		 * Returns the value of the cost of path
+		 * @return uint32_t Cost of path
+		 */
+		inline uint32_t getPathCost() const { return be32toh(getStpConfHeader()->pathCost); }
+
+		/**
+		 * Returns the bridge identifier
+		 * @return uint64_t Bridge identifier
+		 */
+		inline uint64_t getBridgeId() const { return be64toh(getStpConfHeader()->bridgeId); }
+
+		/**
+		 * Returns the priority of bridge
+		 * @return uint16_t Priority of bridge
+		 */
+		inline uint16_t getBridgePriority() const { return be16toh(getStpConfHeader()->bridgeId) & 0xf000; }
+
+		/**
+		 * Returns the system identifier extension of bridge
+		 * @return uint16_t System extension of bridge
+		 */
+		inline uint16_t getBridgeSystemIDExtension() const { return be16toh(getStpConfHeader()->bridgeId) & 0x0fff; }
+
+		/**
+		 * Returns the system identifier of bridge
+		 * @return pcpp::MacAddress System identifier of bridge
+		 */
+		inline pcpp::MacAddress getBridgeSystemID() const { return IDtoMacAddress(getBridgeId()); }
+
+		/**
+		 * Returns the port identifier
+		 * @return uint16_t Port identifier
+		 */
+		inline uint16_t getPortId() const { return be16toh(getStpConfHeader()->portId); }
+
+		/**
+		 * Returns age of the BPDU message
+		 * @return double Age of BPDU
+		 */
+		inline double getMessageAge() const { return be16toh(getStpConfHeader()->msgAge) / 256.0; }
+
+		/**
+		 * Returns maximum age of the BPDU message
+		 * @return double Maximum age of BPDU
+		 */
+		inline double getMaximumAge() const { return be16toh(getStpConfHeader()->maxAge) / 256.0; }
+
+		/**
+		 * Returns the BPDU transmission interval
+		 * @return double Value of the transmission interval
+		 */
+		inline double getTransmissionInterval() const { return be16toh(getStpConfHeader()->helloTime) / 256.0; }
+
+		/**
+		 * Returns the delay for STP message
+		 * @return double Value of the forward delay
+		 */
+		inline double getForwardDelay() const { return be16toh(getStpConfHeader()->forwardDelay) / 256.0; }
```

Comment:
@seladb I don't understand too because I can't find any document/information about it but when I checked the Wireshark implementation [here](https://gitlab.com/wireshark/wireshark/-/blob/master/epan/dissectors/packet-bpdu.c#L609) It uses this logic. I will double check for a detailed information/documentation but an additional pair of eyes can be very helpful 😊.

Reply:
I have no idea why the Wireshark implementation divides by 256... 

I tried to remove both `be16toh(...)` and `/ 256.0` and all tests seem to pass 😃

Reply:
Looks like I found the clear statement [here](https://www.ieee802.org/1/files/public/docs2009/aq-seaman-merged-bpdu-encoding-0509.pdf) at 14.2.8 Encoding of Timer Values section. (It is actually a draft but the original/approved documentation states the same encoding behind a paywall 😒). Every time field is a multiply of 1/256 so the division should stay.

Reply:
So the return value is in seconds, is that correct?
In that case I'd mention in each method doxygen documentation that the return value is in seconds

Reply:
After the division, the value is in seconds. 

I'm not sure the removing both of them is always equal 🤔 but let's remove for now. If it is wrong probably someone will notice and open an issue